src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1467,79 +1467,78 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
       rvd = rvd.orderedJoinDistinct(right.rvd, "inner", joiner, rvd.typ))
   }
 
-  def makeKT(rowExpr: String, entryExpr: String, keyNames: Array[String] = Array.empty, seperator: String = "."): Table = {
-    requireColKeyString("make table")
+  def makeTable(separator: String = "."): Table = {
+    requireColKeyString("make_table")
+    requireUniqueSamples("make_table")
 
-    val vSymTab = Map(
-      "global" -> (0, globalType),
-      "va" -> (1, rowType))
-    val vEC = EvalContext(vSymTab)
-    val vA = vEC.a
-
-    val (vNames, vTypes, vf) = Parser.parseNamedExprs(rowExpr, vEC)
+    val sampleIds = stringSampleIds
 
-    val gSymTab = Map(
-      "global" -> (0, globalType),
-      "va" -> (1, rowType),
-      "sa" -> (2, colType),
-      "g" -> (3, entryType))
-    val gEC = EvalContext(gSymTab)
-    val gA = gEC.a
-
-    val (gNames, gTypes, gf) = Parser.parseNamedExprs(entryExpr, gEC)
-
-    val sig = TStruct(((vNames, vTypes).zipped ++
-      stringSampleIds.flatMap { s =>
-        (gNames, gTypes).zipped.map { case (n, t) =>
-          (if (n.isEmpty)
-            s
-          else
-            s + seperator + n, t)
-        }
-      }).toSeq: _*)
+    val ttyp = TableType(
+      TStruct(
+        matrixType.rowType.fields.map { f => f.name -> f.typ } ++
+          sampleIds.flatMap { s =>
+            matrixType.entryType.fields.map { f =>
+              val newName = if (f.name == "") s else s + separator + f.name
+              newName -> f.typ
+            }
+          }: _*),
+      matrixType.rowKey,
+      matrixType.globalType)
 
     val localNSamples = numCols
-    val localColValuesBc = colValues.broadcast
     val localRVRowType = rvRowType
-    val globalsBc = globals.broadcast
     val localEntriesIndex = entriesIndex
+    val localEntriesType = localRVRowType.types(entriesIndex).asInstanceOf[TArray]
+    val localEntryType = matrixType.entryType
```

Comment:
Is there ever a case where `localEntriesType` is not just `TArray(matrixType.entryType)`? Or is this just a way to type check it's a TArray? If so, maybe add an assert that the `matrixType.entryType == localEntriesType.elementType`.

Reply:
I cast because I need to call isElementDefined loadElement down below.  Yes, right now it is always TArray(entryType), however, it seems safer to get the type from the rvRowType because that's the actual type in case it changes.